### ISSUE TITLE
Shiny Render Failing

### ISSUE THREAD

**Issue by GrayAlex49** [malcolmbarrett/ggdag#55]
I'm trying to render a dag in shiny and while it renders it in the console just fine, it errors out in shiny. In my reproducible example I just get the error but in a production app it ends up segfaulting shiny server and taking everything down with it. My colleague is able to run the same code with all the same versions of everything on a mac, I am on a pc. but I've also had trouble in a docker environment (I'm working on a reproducible example of that but having trouble getting daggity installed). 

```
library(shiny)
library(dagitty)
library(ggdag)


ui <- fluidPage(
  plotOutput("plot")
)

server <- function(input, output, session) {

  bigger_dag <- dagify(y ~ x + a + b,
                       x ~ a + b,
                       exposure = "x",
                       outcome = "y")

  print(bigger_dag)
  output$plot <- renderPlot({
    
    bigger_dag %>% 
      tidy_dagitty() %>%
      ggplot(aes(x = x, y = y, xend = xend, yend = yend)) +
      geom_dag_point() +
      geom_dag_edges() +
      geom_dag_text(col = "white") +
      theme_dag()
    
    
  })
}

shinyApp(ui, server)
```

`Warning: Error in context_eval: <string conversion failed>`

```
> sessionInfo()
R version 4.0.5 (2021-03-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggdag_0.2.3   ggplot2_3.3.3 dagitty_0.3-1 shiny_1.6.0  

loaded via a namespace (and not attached):
 [1] bslib_0.2.4       tidyselect_1.1.1  shinyjs_2.0.0     purrr_0.3.4      
 [5] V8_3.4.2          colorspace_2.0-1  vctrs_0.3.8       generics_0.1.0   
 [9] htmltools_0.5.1.1 utf8_1.2.1        blob_1.2.1        rlang_0.4.11     
[13] jquerylib_0.1.4   later_1.2.0       pillar_1.6.0      glue_1.4.2       
[17] withr_2.4.2       DBI_1.1.1         bit64_4.0.5       lifecycle_1.0.0  
[21] stringr_1.4.0     munsell_0.5.0     gtable_0.3.0      memoise_2.0.0    
[25] fastmap_1.1.0     httpuv_1.6.1      curl_4.3.1        fansi_0.4.2      
[29] tidygraph_1.2.0   Rcpp_1.0.6        xtable_1.8-4      promises_1.2.0.1 
[33] scales_1.1.1      cachem_1.0.4      jsonlite_1.7.2    mime_0.10        
[37] bit_4.0.4         digest_0.6.27     stringi_1.5.3     dplyr_1.0.6      
[41] grid_4.0.5        tools_4.0.5       sass_0.3.1        magrittr_2.0.1   
[45] RSQLite_2.2.7     tibble_3.1.1      crayon_1.4.1      tidyr_1.1.3      
[49] pkgconfig_2.0.3   MASS_7.3-53.1     ellipsis_0.3.2    rstudioapi_0.13  
[53] assertthat_0.2.1  httr_1.4.2        R6_2.5.0          boot_1.3-27      
[57] igraph_1.2.6      compiler_4.0.5 
```

**Comment by malcolmbarrett** [comment]
This seems to be related to https://github.com/rstudio/shiny/issues/3289. dagitty uses the V8 package, which seems to crash on certain Windows machines, seemingly R > 4.0. Since this isn't a ggdag issue by the looks of this, I'll direct you that way and close this. If you find out ggdag is having an issue on top of that, please feel free to re-open.

Thanks for filing!

**Comment by GrayAlex49** [comment]
I agree, thank you. Rolling back to R 3.6.3 resolves this issue.

**Comment by jeroen** [comment]
This should be fixed in V8 3.6.0, now on CRAN. The problem was that we were hitting the JS stack limit. We have now relaxed the stack limit and also provide better error messages when we do hit it.
